frontends/uhdm/UhdmAst.cc

```diff
@@ -242,8 +248,44 @@ static void add_or_replace_child(AST::AstNode* parent, AST::AstNode* child) {
 			*it = child;
 			return;
 		}
+	} else if (child->type == AST::AST_INITIAL) {
+		// Special case for initials
+		// Ensure that there is only one AST_INITIAL in the design
+		// And there is only one AST_BLOCK inside that initial
+		// Copy nodes from child initial to parent initial
+		auto initial_node_it = find_if(parent->children.begin(), parent->children.end(),
+			[] (AST::AstNode *node) {return (node->type == AST::AST_INITIAL);} );
+		if (initial_node_it != parent->children.end()) {
+			AST::AstNode* initial_node = *initial_node_it;
+
+			log_assert(!(initial_node->children.empty()));
+			log_assert(initial_node->children[0]->type == AST::AST_BLOCK);
+			log_assert(!(child->children.empty()));
+			log_assert(child->children[0]->type == AST::AST_BLOCK);
+
+			AST::AstNode* block_node = initial_node->children[0];
+			AST::AstNode* child_block_node = child->children[0];
+
+			// Place the contents of child initial node inside the parent initial
+			// And place the contents of child block node inside parent block
+			for (auto initial_child : child->children) {
+				if (initial_child == child_block_node) {
+					for (auto child_block_child : child_block_node->children)
+						block_node->children.push_back(child_block_child->clone());
+				} else {
+					initial_node->children.push_back(initial_child->clone());
+				}
```

Comment:
With recent changes, here we already know, that there is AST_BLOCK as first child of ``child`` node, so I think this if is unnecessary, and we can just always copy nodes.

If it is not the case, here we are comparing pointers, but we shouldn't do that. This can introduce 2 ``AST_BLOCK`` inside ``AST_INITIAL``.

Reply:
I removed the if statement. Now we first clone the contents of child block into parent block and then proceed with copying other children nodes of the child node into parent initial node.